Add language around voting members and affirmation (#13)
django/dsf-bylaws: bylaws.md

Upon any resignation or termination of the membership of any Member, all rights of membership, including all related voting rights, of such Member shall be terminated. After a resignation or termination of the membership of any Member, such Member may be reconsidered for membership only in accordance with Section 3.2 of these bylaws. 

> ### 3.7. Voting Affirmation

Reviewer: ```suggestion
### 3.7. Voting Members
```

Tweak, but I think this makes slightly more sense as a section header.

#### 3.7.1

> Members with voting rights must affirm to the Foundation in writing each year that they intend to vote in that year's elections. Such affirmation must be received by the Foundation before the opening of the relevant election. Only Members who have affirmed their intent to vote, and whose membership and voting rights are otherwise in good standing, shall be considered "**Voting Members**" for purposes of elections and quorum calculations under these bylaws.

Reviewer: ```suggestion
Members of any class with voting rights must affirm to the Foundation in writing each year that they intend to vote in that year's elections. Such affirmation must be received by the Foundation before the opening of the relevant election. Only Members who have affirmed their intent to vote, and whose membership and voting rights are otherwise in good standing, shall be considered "**Voting Members**" for purposes of elections and quorum calculations under these bylaws.
```

Clarify that this applies to any current _or future_ membership class that includes voting rights.

This opens up a can of worms: do corporate members have voting rights? I'll address that in a separate comment.

> Members of any class with voting rights must affirm to the Foundation in writing each year that they intend to vote in that year's elections. Such affirmation must be received by the Foundation before the opening of the relevant election. Only Members who have affirmed their intent to vote, and whose membership and voting rights are otherwise in good standing, shall be considered "**Voting Members**" for purposes of elections and quorum calculations under these bylaws.

Reviewer 2: As an alternative, perhaps something like this could work (and remove Section 3.7.2 entirely), **updated 2026-07-23 to include either of the two most recent elections:**

```suggestion
Members of any class with voting rights must affirm to the Foundation in writing each year that they intend to vote in that year's elections. Members may affirm such intent by (a) having voted in either of the two (2) most recent annual elections, or (b) submitting a written affirmation to the Foundation before the opening of the relevant election. Only Members who have affirmed their intent to vote, and whose membership and voting rights are otherwise in good standing, shall be considered "**Voting Members**" for purposes of elections and quorum calculations under these bylaws.
```

#### 3.7.2

> The Board may waive the affirmation requirement for any Member who voted in the most recent preceding election. Such Members shall be considered Voting Members without a new affirmation unless they notify the Foundation in writing that they do not intend to vote.

Reviewer 3: What would the mechanism be for waiving this requirement? What would the criteria be for such a waiver?

Author: The criteria are in the wording - that they voted in the last election. If you mean when would we do this, I think we could do it always, so in reality you only have to affirm once every couple of elections. One thing to note is that sometimes the SC elections and board elections can happen close together so it might be a bit annoying to need to affirm twice.

The mechanism I'm less sure of, but that's mostly because I don't really know the mechanism for affirmation in the first place, we would need some way to track it and ideally automate it all.

Reviewer: Broadly the way I see this working is that some time prior to the election, the administrator does some things:

-  Collect a list of everyone who voted in last year's election; that's our beginning Voting Members list for this year. 
- Everyone on that list gets an email saying: "hi, you voted last year, so automatically registered to vote this year. If you don't want to vote, email us to be removed." (Anyone who emails is removed.)
-  Every member who _didn't_ vote last year gets an email: "hi, in order to vote in the upcoming election, you need to register to vote. click here." (This could link to a google form, whatever). Anyone who registers by some deadline is is now a voting member.
- Election begins: everyone on the list gets a link to the ballot.

This absolutely doesn't need to be in the bylaws, but it's worth writing down in some procedure guide somewhere.

Reviewer 4: The use of "may" might be confusing if that's the way it's going to be implemented. If the idea is that everyone who voted in the last election to not need to affirm, it might as well be explicit: 

```suggestion
The Board waives the affirmation requirement for any Member who voted in the most recent preceding election. Such Members shall be considered Voting Members without a new affirmation unless they notify the Foundation in writing that they do not intend to vote.
```

> The Board may waive the affirmation requirement for any Member who voted in the most recent preceding election. Such Members shall be considered Voting Members without a new affirmation unless they notify the Foundation in writing that they do not intend to vote.

Reviewer: ```suggestion
The Board will waive the affirmation requirement for any Member who voted in the most recent preceding election. Such Members shall be considered Voting Members without a new affirmation unless they notify the Foundation in writing that they do not intend to vote.
```

I'm unsure about this change, we should probably discuss.

If it's "may": the Board will need to take affirmative action each year (e.g. a vote) to waive affirmation for people who voted last year. If it's "will", this would be assumed to happen automatically, and we don't have to do anything. There are pros and cons to each approach!

Reviewer 2: I agree with this, and proposed an alternative in 3.7.1 to remove the "waiving" requirement altogether.

I'm also not clear on the value of Members being able to notify the Foundation that they do not intend to vote. Is this something we expect folks to take advantage of, and are the logistics of tracking "no" RSVPs worth the effort?

> The Board may waive the affirmation requirement for any Member who voted in the most recent preceding election. Such Members shall be considered Voting Members without a new affirmation unless they notify the Foundation in writing that they do not intend to vote.

Reviewer 5: what is the benefit of having this clause? the way I see it:

- it increases the risk of voting members not actually voting vs removing the clause entirely
- it increases operational overhead
- it can be seen as unfair to members who are required to click a link - if you were already a voting member and forgot about an election, you can still vote at the last minute. if you forgot about the election and needed to click a link, then you cannot, despite the same circumstances in both.

why not just remove this clause and require all members to confirm intention to vote for every ballot?

Reviewer 6: It’s true that requiring every member to affirm before every election would create a clearer and more consistent process. However, the board’s concern was that adding that friction could reduce participation at a time when some elections are already barely meeting quorum.

The question we were trying to address was what would happen if fewer than 50% of the required voting members participated, as required by the bylaws. This clause gives the board some flexibility to avoid unintentionally shrinking the voting pool by requiring recent, active voters to complete another administrative step.

The goal is to let members who have recently participated continue voting with less friction, rather than risk excluding them because they missed an email asking them to click an affirmation link. At the same time, the affirmation process gives members who have not been active a clear way to opt back in. If one election cycle is too narrow a measure of recent participation, we could consider rewording the clause to cover members who voted within the previous two years.

Reviewer 2: I agree with the overall goal: We want to reduce friction for active voters while naturally letting truly inactive members roll off.

FWIW, I also believe this clause adds unnecessary complication and could be removed, if we interpret voting itself as a way to affirm intention to vote in future elections.

I've broadened my proposed edit above to include either of the two most recent annual elections:

https://github.com/django/dsf-bylaws/pull/13#discussion_r3553346089

#### 3.7.3

> The Foundation shall make reasonable efforts to notify all eligible Members of the affirmation requirement and the deadline for affirmation at least thirty (30) days before the opening of the relevant election.

Reviewer: ```suggestion
The Foundation shall make reasonable efforts to notify all eligible Members of the affirmation requirement and the deadline for affirmation before the opening of the relevant election.
```

I don't love hardcoding deadlines into bylaws.
